source/material_model/rheology/elasticity.cc

```diff
@@ -139,8 +141,29 @@ namespace aspect
         if (this->convert_output_to_years())
           fixed_elastic_time_step *= year_in_seconds;
 
+        // When using the visco_plastic material model,
+        // make sure that no damping is applied.
+        // If a damper is used, the current implementation in visco_plastic
+        // is not correct. The way the effective viscosity is computed is only valid
+        // in the absence of a damper and for a yield stress that can be expressed as
+        // some value of the second invariant of the deviatoric stress.
+        // The visco_plastic and viscoelastic material models also require an operator splitting
+        // step to update the stresses stored, and a discontinuous element to accommodate discontinuous
+        // strain rates that feed into the stored stresses.
+        if (Plugins::plugin_type_matches<MaterialModel::ViscoPlastic<dim>>(this->get_material_model()) ||
+            Plugins::plugin_type_matches<MaterialModel::Viscoelastic<dim>>(this->get_material_model()))
+          AssertThrow(elastic_damper_viscosity == 0. &&
+                      this->get_parameters().use_operator_splitting &&
+                      this->get_parameters().use_discontinuous_composition_discretization,
+                      ExcMessage("The visco-plastic material model with elasticity enabled requires "
+                                 "an elastic timestep equal to the computational timestep, operator splitting "
+                                 "no timestep stabilization, no elastic damping and the use of discontinuous "
+                                 "elements for composition."));
+
         AssertThrow(this->get_parameters().enable_elasticity == true,
-                    ExcMessage ("Material model Viscoelastic only works if 'Enable elasticity' is set to true"));
+                    ExcMessage("Material model Viscoelastic only works if 'Enable elasticity' is set to true"));
```

Comment:
elasticity.cc is already used for ViscoPlastic. 
rheology/visco_plastic.cc reads in a parameter `use_elasticity = prm.get_bool ("Include viscoelasticity");` and only when `use_elasticity` is true, `elastic_rheology` is initialized etc. I think that's why it works. I've found having to set `Enable elasticity` and `Include viscoelasticity` in two different places a bit unintuitive, but changing it would be a separate PR. We can discuss it with John.